component/cargo/xc/src/lib/lib.rs

```diff
@@ -52,6 +53,14 @@ pub fn app<I: io::Read, O: io::Write>(
 
                 Ok(())
             }
+            NULL_SEPARATED => {
+                let mut separator = "";
+                buffer.lines().for_each(move |line| {
+                    let _ = write!(out, "{}{}", separator, line);
+                    separator = "\x00";
```

Comment:
I think this assignment can be [avoided](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold) as well.